torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -319,8 +319,10 @@ class TORCH_API ProcessGroupNCCL : public Backend {
     // to the store.
     c10::intrusive_ptr<Store> store_;
 
-    // Store a reference to NCCL collective's outputs, used by result and to
-    // give a more descriptive message when representing the Work as a string.
+    // Store a reference to NCCL collective's outputs and inputs, used by
+    // result, to give a more descriptive message when representing the Work as
+    // a string, and for Flight Recorder.
+    std::shared_ptr<std::vector<at::Tensor>> inputs_;
```

Comment:
I can see that FR needs `inputs_`, but it seems we can satisfy it by passing `inputs` to the `initWork` function (where `record()` occurs). But does that mean we'd also need to create an `inputs_` field in the `Work` object? I'd prefer having as few reference to tensor as possible. Or do I miss some other purpose?

Reply:
oops that was leftover from one of my previous attempts.  i just removed it.